### ISSUE TITLE
Not working under Windows due to fork() not implemented.

### ISSUE THREAD

**Issue by ghost** [TimothyYe/DynDNS#2]
Hi, thanks for this tool.
I'm using windows, which this script won't work: 
```shell
D:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/daemons-1.2.4/lib/daemons/daemonize.rb:10:in `fork': fork() function is unimplemented on this machine (NotImplementedError)
        from D:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/daemons-1.2.4/lib/daemons/daemonize.rb:10:in `safefork'
        from D:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/daemons-1.2.4/lib/daemons/daemonize.rb:45:in `call_as_daemon'
        from D:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/daemons-1.2.4/lib/daemons/application.rb:270:in `start_proc'
        from D:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/daemons-1.2.4/lib/daemons/application.rb:296:in `start'
        from D:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/daemons-1.2.4/lib/daemons/controller.rb:56:in `run'
        from D:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/daemons-1.2.4/lib/daemons.rb:197:in `block in run_proc'
        from D:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/daemons-1.2.4/lib/daemons/cmdline.rb:92:in `catch_exceptions'
        from D:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/daemons-1.2.4/lib/daemons.rb:196:in `run_proc'
        from DynDNS.rb:15:in `<main>'
```
Is there an easy way to bypass this error and make it working?

**Comment by TimothyYe** [comment]
Hi, you may try to disable this daemon module.

Comment out these lines in DynDNS.rb:

1. # require 'daemons' at line 11

2. # Daemons.run_proc("DynDNS") do  at line 15

3. And the last "end" at line 45.

Then, try to run it again...

**Comment by hxhxd** [comment]
@TimothyYe 
Hi, thanks for reply.
I've tried that, it worked, yet I still want to damonize it.
Googled it, some suggest using win32-service, I checked its document, not a easy solution like this, so I didn't try it.

At last, I removed windows native version of Ruby, installed cygwin version.
There's a logger problem which the log file won't update, but the script works.
```shell
log writing failed. closed stream
```
I've tried change the log file's permission, aside from set it 777 in cygwin, no luck.
